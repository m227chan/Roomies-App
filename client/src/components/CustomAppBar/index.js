import React from 'react';
import Typography from "@material-ui/core/Typography";
import Link from '@material-ui/core/Link';
import history from '../Navigation/history';

import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Button from '@material-ui/core/Button';

const CustomAppBar = () => {

  return (
    <AppBar position="static">
      <Toolbar>

        <Button color="inherit">
          <Link
            color="inherit"
            style={{ cursor: "pointer" }}
            onClick={() => history.push('/Room')}
          >
            <Typography variant="h6" color="inherit">
              Room
            </Typography>
          </Link>
        </Button>

        <Button color="inherit" data-testid = 'Navigation Button to Room'>
          <Link
            color="inherit"
            style={{ cursor: "pointer" }}
            onClick={() => history.push('/Calendar')}
          >
            <Typography variant="h6" color="inherit">
              Calendar
            </Typography>
          </Link>
        </Button>

        <Button color="inherit">
          <Link
            color="inherit"
            style={{ cursor: "pointer" }}
            onClick={() => history.push('/Expenses')}
          >
            <Typography variant="h6" color="inherit">
              Expenses
            </Typography>
          </Link>
        </Button>

        <Button color="inherit">
          <Link
            color="inherit"
            style={{ cursor: "pointer" }}
            onClick={() => history.push('/Grocery')}
          >
            <Typography variant="h6" color="inherit">
              Grocery
            </Typography>
          </Link>
        </Button>

        <Button color="inherit">
          <Link
            color="inherit"
            style={{ cursor: "pointer" }}
            onClick={() => history.push('/Settings')}
          >
            <Typography variant="h6" color="inherit">
              Settings
            </Typography>
          </Link>
        </Button>

      </Toolbar>
    </AppBar>
  );
}

export default CustomAppBar;