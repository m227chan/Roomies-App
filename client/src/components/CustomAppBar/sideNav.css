
.navGrid {
    display: flex;
    height: 100%;
    flex-direction: column;
    justify-content: space-around;
    align-items: stretch;
    background-color: var(--color-primary);
    color: var(--color-bg);
}

.navMiddle{
    padding: 0% 5%;
}
.navHeader {
    display: flex;
    justify-content: center;
    align-items: center;
    margin: 0% 10%;
    
  }

  .navAvatar {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
  }

  .navBottom{
    display: flex;
    margin: 0 10%;
    flex-direction: row;
    justify-content: space-evenly;
    align-items: center;
  }

  .navIcon {
    color: var(--color-bg);
  }
  
  