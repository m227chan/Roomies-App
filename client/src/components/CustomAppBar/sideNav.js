// Importing react and files
import * as React from "react";
import "./sideNav.css";
import history from "../Navigation/history";

// Importing material
import Box from "@mui/material/Box";
import Drawer from "@mui/material/Drawer";
import List from "@mui/material/List";
import Typography from "@mui/material/Typography";
import Divider from "@mui/material/Divider";
import ListItem from "@mui/material/ListItem";
import ListItemButton from "@mui/material/ListItemButton";
import ListItemIcon from "@mui/material/ListItemIcon";
import { Grid } from "@mui/material";
import Avatar from "@mui/material/Avatar";
import { grey } from "@mui/material/colors";
import { createTheme, ThemeProvider } from "@mui/material/styles";

// Importing icons
import SettingsIcon from "@mui/icons-material/Settings";
import LogoutIcon from "@mui/icons-material/Logout";
import CalendarMonthIcon from "@mui/icons-material/CalendarMonth";
import ShoppingCartIcon from "@mui/icons-material/ShoppingCart";
import CurrencyExchangeIcon from "@mui/icons-material/CurrencyExchange";
import HomeWorkIcon from "@mui/icons-material/HomeWork";

// Sets the width
const drawerWidth = 240;

// Sets the theme for the file
const theme = createTheme({
  typography: {
    fontFamily: "Tajawal",
    subtitle1: {
      fontSize: 18,
    },
  },
});

// Component for the top of the Nav
const NavProfile = () => {
  return (
    <Grid item class="navAvatar">
      {" "}
      <Avatar
        sx={{
          bgcolor: grey[200],
          color: "#02473B",
          width: 60,
          height: 60,
        }}
      >
        <Typography variant="h4">Z</Typography>
      </Avatar>
      <Typography
        style={{ paddingTop: "10%", fontWeight: "bold" }}
        variant="h6"
      >
        {" "}
        Zach Zammit
      </Typography>
    </Grid>
  );
};
// Component for the middle of the Nav
const NavMenu = () => {
  return (
    <Grid item class="navMiddle">
      <Grid class="navHeader">
        <Grid item xs={4}>
          <Typography variant="p" sx={{ fontWeight: "light" }}>
            Home{" "}
          </Typography>
        </Grid>
        <Grid item xs={9}>
          <Divider style={{ background: "#FFFFFF" }} />
        </Grid>
      </Grid>
      <List sx={{ margin: "5% 0%" }}>
        {["Room"].map((text, index) => (
          <ListItem key={text}>
            <ListItemButton>
              <ListItemIcon style={{ minWidth: "40px" }}>
                <HomeWorkIcon sx={{ color: "#FFFFFF" }} />
              </ListItemIcon>
              <Typography
                variant="subtitle1"
                sx={{ fontWeight: "regular" }}
                onClick={() => history.push("/Room")}
              >
                {text}
              </Typography>
            </ListItemButton>
          </ListItem>
        ))}
      </List>
      <Grid class="navHeader">
        <Grid item xs={4}>
          <Typography variant="p" sx={{ fontWeight: "light" }}>
            Spending{" "}
          </Typography>
        </Grid>
        <Grid item xs={8}>
          <Divider style={{ marginLeft: "12%", background: "#FFFFFF" }} />
        </Grid>
      </Grid>
      <List sx={{ margin: "5% 0%" }}>
        {["Expenses", "Grocery"].map((text, index) => (
          <ListItem key={text}>
            <ListItemButton>
              <ListItemIcon style={{ minWidth: "40px" }}>
                {index % 2 === 0 ? (
                  <CurrencyExchangeIcon sx={{ color: "#FFFFFF" }} />
                ) : (
                  <ShoppingCartIcon sx={{ color: "#FFFFFF" }} />
                )}
              </ListItemIcon>
              <Typography
                variant="subtitle1"
                onClick={() => history.push("/" + text)}
              >
                {text}
              </Typography>
            </ListItemButton>
          </ListItem>
        ))}
      </List>
      <Grid class="navHeader">
        <Grid item xs={4}>
          <Typography variant="p" sx={{ fontWeight: "light" }}>
            Events
          </Typography>
        </Grid>
        <Grid item xs={10}>
          <Divider style={{ background: "#FFFFFF" }} />
        </Grid>
      </Grid>
      <List sx={{ margin: "5% 0%" }}>
        {["Calendar"].map((text, index) => (
          <ListItem key={text}>
            <ListItemButton>
              <ListItemIcon style={{ minWidth: "40px" }}>
                <CalendarMonthIcon sx={{ color: "#FFFFFF" }} />
              </ListItemIcon>
              <Typography
                variant="subtitle1"
                onClick={() => history.push("/Calendar")}
              >
                {text}
              </Typography>
            </ListItemButton>
          </ListItem>
        ))}
      </List>
    </Grid>
  );
};
// Component for the bottom of the Nav
const NavBottom = () => {
  return (
    <Grid item class="navBottom">
      <SettingsIcon
        onClick={() => history.push("/Settings")}
        sx={{
          "&:hover": {
            cursor: "pointer",
            color: "#D2EADA",
          },
        }}
      />
      <Typography sx={{ fontWeight: "bold" }}>|</Typography>
      <LogoutIcon
        onClick={() => history.push("/Settings")}
        sx={{
          "&:hover": {
            cursor: "pointer",
            color: "#D2EADA",
          },
        }}
      />
    </Grid>
  );
};

export default function SideNav() {
  return (
    <ThemeProvider theme={theme}>
      <Box sx={{ display: "flex" }}>
        <Drawer
          sx={{
            width: drawerWidth,
            flexShrink: 0,
            "& .MuiDrawer-paper": {
              width: drawerWidth,
              boxSizing: "border-box",
            },
          }}
          variant="permanent"
          anchor="left"
        >
          <Grid class="navGrid">
            <NavProfile />
            <NavMenu />
            <NavBottom />
          </Grid>
        </Drawer>
      </Box>
    </ThemeProvider>
  );
}
