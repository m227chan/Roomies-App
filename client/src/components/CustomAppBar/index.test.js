import React from 'react';
import renderer from 'react-test-renderer';
import {render} from '@testing-library/react';
import CustomAppBar from './index';

describe('CustomAppBar', () => {

    it('renders correctly', () => {
        const tree = renderer
            .create(<CustomAppBar />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    // describe('RestaurantList', () => {
    //     it('loads restaurants on first render', () => {
    //         const loadRestaurants = jest.fn().mockName('loadRestaurants');
    //         render(<RestaurantList loadRestaurants={loadRestaurants} />);
    //         expect(loadRestaurants).toHaveBeenCalled();
    //     });
    // });

});