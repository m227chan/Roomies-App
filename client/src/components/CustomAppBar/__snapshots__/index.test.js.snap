// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomAppBar renders correctly 1`] = `
<header
  className="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
>
  <div
    className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        <a
          className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorInherit"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          style={
            Object {
              "cursor": "pointer",
            }
          }
        >
          <h6
            className="MuiTypography-root MuiTypography-h6 MuiTypography-colorInherit"
          >
            Room
          </h6>
        </a>
      </span>
    </button>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        <a
          className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorInherit"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          style={
            Object {
              "cursor": "pointer",
            }
          }
        >
          <h6
            className="MuiTypography-root MuiTypography-h6 MuiTypography-colorInherit"
          >
            Calendar
          </h6>
        </a>
      </span>
    </button>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        <a
          className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorInherit"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          style={
            Object {
              "cursor": "pointer",
            }
          }
        >
          <h6
            className="MuiTypography-root MuiTypography-h6 MuiTypography-colorInherit"
          >
            Expenses
          </h6>
        </a>
      </span>
    </button>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        <a
          className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorInherit"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          style={
            Object {
              "cursor": "pointer",
            }
          }
        >
          <h6
            className="MuiTypography-root MuiTypography-h6 MuiTypography-colorInherit"
          >
            Grocery
          </h6>
        </a>
      </span>
    </button>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        <a
          className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorInherit"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          style={
            Object {
              "cursor": "pointer",
            }
          }
        >
          <h6
            className="MuiTypography-root MuiTypography-h6 MuiTypography-colorInherit"
          >
            Settings
          </h6>
        </a>
      </span>
    </button>
  </div>
</header>
`;
